### PR TITLE
Cli

### PR DESCRIPTION
**Pull request by guyst16**
Add, Delete, Reload and List

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+![ARMadillo-logo](https://user-images.githubusercontent.com/100173467/203181147-e1d61fb0-426b-44f4-8a9b-08b7c6793450.png)
```
Comment (doronkg) [inline README.md:1]: Remove this part as the logo will be added at `docs` branch

File: utils.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import configparser
+import os.path
+
+armadillo_dir_path = os.path.join(os.path.expanduser("~"),".armadillo")
```
Comment (doronkg) [inline utils.py:4]:
```suggestion
armadillo_dir_path = os.path.join(os.path.expanduser("~"), ".armadillo")
```

File: utils.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+# Return config parser from file
+def get_config(file_path):
+    config = configparser.ConfigParser()
+    config.read(file_path)
+    return config
+
+# Save to file
+def save_to_file(config,
+                 file_path):
```
Comment (doronkg) [inline utils.py:18]: Why the new line?

File: armadillo_cli.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import armadillo_cli_add
+import armadillo_cli_list
+import armadillo_cli_delete
+import armadillo_cli_reload
+
+if not os.path.exists(utils.armadillo_dir_path):
+    os.mkdir(utils.armadillo_dir_path)
+
+app = typer.Typer()
+app.add_typer(armadillo_cli_add.app, name="add", help="Add new policy      (policy|policy-dns|policy-dpi)")
```
Comment (doronkg) [inline armadillo_cli.py:14]:
```suggestion
app.add_typer(armadillo_cli_add.app, name="add", help="Add new policy (policy|policy-dns|policy-dpi)")
```

File: armadillo_cli.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+app.add_typer(armadillo_cli_list.app, name="list", help="List all policies   (policy|policy-dns|policy-dpi)")
```
Comment (doronkg) [inline armadillo_cli.py:15]:
```suggestion
app.add_typer(armadillo_cli_list.app, name="list", help="List all policies (policy|policy-dns|policy-dpi)")
```

File: armadillo_cli_add.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
+import typer
+import utils
+
+app              = typer.Typer()
+SECTION_NAME     = "policy"
+
+# Generate new section name
+def get_next_section_name(config):
+    new_section = ""
```
Comment (doronkg) [inline armadillo_cli_add.py:9]:
```suggestion
    new_section = SECTION_NAME
```

And remove it from the following lines, this way you avoid assigning this var twice

File: armadillo_cli_add.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+    filepath = utils.dpi_policy_path
+
+    config            = utils.get_config(filepath)
+    next_section_name = get_next_section_name(config)
+    config.add_section(next_section_name)
+    config.set(next_section_name, 'string', string)
+    config.set(next_section_name, 'enabled', str(int(enabled)))
+    utils.save_to_file(config, filepath)
+
```
Comment (doronkg) [inline armadillo_cli_add.py:66]:
```suggestion
```

File: armadillo_cli_add.py [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+import utils
+
+app              = typer.Typer()
+SECTION_NAME     = "policy"
+
+# Generate new section name
+def get_next_section_name(config):
+    new_section = ""
+    if len(config.sections()) != 0:
+        new_section = SECTION_NAME+'-'+str(int(config.sections()[-1].split('-')[1])+1)
```
Comment (doronkg) [inline armadillo_cli_add.py:11]:
```suggestion
        new_section = '_' + str(int(config.sections()[-1].split('-')[1]) + 1)
```

File: armadillo_cli_add.py [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    else:
+        new_section = SECTION_NAME+'-1'
```
Comment (doronkg) [inline armadillo_cli_add.py:13]:
```suggestion
        new_section += '_1'
```

File: armadillo_cli_delete.py [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+import utils
+
+app = typer.Typer()
+
+@app.command()
+def policy(policy_id: str = typer.Option(...,help="The policy id to delete e.g: --policy-id=policy-1")):
+    filepath = utils.policy_file_path
+    config   = utils.get_config(filepath)
+    config.remove_section(policy_id)
+    utils.save_to_file(config,filepath)
```
Comment (doronkg) [inline armadillo_cli_delete.py:11]:
```suggestion
    utils.save_to_file(config, filepath)
```

File: armadillo_cli_delete.py [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+@app.command()
+def policy_dns(policy_id: str = typer.Option(...,help="The policy id to delete e.g: --policy-id=policy-1")):
+    filepath = utils.dns_policy_path
+    config   = utils.get_config(filepath)
+    config.remove_section(policy_id)
+    utils.save_to_file(config,filepath)
```
Comment (doronkg) [inline armadillo_cli_delete.py:18]:
```suggestion
    utils.save_to_file(config, filepath)
```

File: armadillo_cli_delete.py [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+
+@app.command()
+def policy_dpi(policy_id: str = typer.Option(...,help="The policy id to delete e.g: --policy-id=policy-1")):
+    filepath = utils.dpi_policy_path
+    config   = utils.get_config(filepath)
+    config.remove_section(policy_id)
+    utils.save_to_file(config,filepath)
```
Comment (doronkg) [inline armadillo_cli_delete.py:25]:
```suggestion
    utils.save_to_file(config, filepath)
```

File: armadillo_cli_delete.py [hunk 13]
```diff
@@ -0,0 +1,29 @@
+import typer
+import utils
+
+app = typer.Typer()
+
+@app.command()
+def policy(policy_id: str = typer.Option(...,help="The policy id to delete e.g: --policy-id=policy-1")):
```
Comment (doronkg) [inline armadillo_cli_delete.py:7]:
```suggestion
def policy(policy_id: str = typer.Option(...,help="The policy id to delete e.g: --policy-id=policy_1")):
```

File: armadillo_cli_delete.py [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+
+@app.command()
+def policy_dns(policy_id: str = typer.Option(...,help="The policy id to delete e.g: --policy-id=policy-1")):
+    filepath = utils.dns_policy_path
+    config   = utils.get_config(filepath)
+    config.remove_section(policy_id)
+    utils.save_to_file(config,filepath)
+
+@app.command()
+def policy_dpi(policy_id: str = typer.Option(...,help="The policy id to delete e.g: --policy-id=policy-1")):
```
Comment (doronkg) [inline armadillo_cli_delete.py:21]:
```suggestion
def policy_dpi(policy_id: str = typer.Option(...,help="The policy id to delete e.g: --policy-id=policy_1")):
```

File: armadillo_cli_list.py [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+        table.append(row)
+
+    return table
+
+@app.command(help="list all policies")
+def policy():
+    filepath = utils.policy_file_path
+    config   = utils.get_config(filepath)
+    table = get_rows(config)
+    print(tabulate(table,headers=["Source Port", "Destination Port", "Source Address", "Destination Address", "Protocol", "Enabled"]))
```
Comment (doronkg) [inline armadillo_cli_list.py:24]:
```suggestion
    print(tabulate(table, headers=["Source Port", "Destination Port", "Source Address", "Destination Address", "Protocol", "Enabled"]))
```

File: armadillo_cli_list.py [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+    config   = utils.get_config(filepath)
+    table = get_rows(config)
+    print(tabulate(table,headers=["Domain", "Enabled"]))
+
+@app.command(help="list all dpi policies")
+def policy_dpi():
+    filepath = utils.dpi_policy_path
+    config   = utils.get_config(filepath)
+    table = get_rows(config)
+    print(tabulate(table,headers=["String", "Enabled"]))
```
Comment (doronkg) [inline armadillo_cli_list.py:38]:
```suggestion
    print(tabulate(table, headers=["String", "Enabled"]))
```

File: armadillo_cli_list.py [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+
```
Comment (doronkg) [inline armadillo_cli_list.py:39]:
```suggestion
```

**Review by doronkg — CHANGES_REQUESTED** [review]
The logic itself works and looks great.

Do notice the comments throughout the code.
Please allign all the .py files with the same convention, whether value assigning is on the same column or not, because sometimes it is and sometimes it isn't.

One of the following options:
```
a = 1
ab = 2
abc = 3
---
a   = 1
ab  = 2
abc = 3
```

File: armadillo_cli.py (hunk 5 above)
Comment (guyst16) [reply armadillo_cli.py:15]: For keep all the commands' args in the same line

**Pull request closed without merging**